### PR TITLE
Refactor: Move classes to internal package

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

Move appropriate classes to internal package. (Fixes [#2864](https://github.com/mapbox/mapbox-navigation-android/issues/2864))

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Move appropriate classes to internal package.

### Implementation

Move appropriate classes to internal package.

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java [hunk 1]
```diff
@@ -279,6 +280,7 @@ public void unsubscribe() {
       navigationViewModel.retrieveIsOffRoute().removeObservers(lifecycleOwner);
       navigationViewModel.retrieveRouteJunctionModelUpdates().removeObservers(lifecycleOwner);
     }
+    ImageCreator.getInstance().shutdown();
```
Comment (kmadsen) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java:283]: @abhishek1508 this on purpose? looks like a legit fix, but surfacing cause it looks unrelated

**Comment by JunDai** [comment]
@abhishek1508 - qq: what's the purpose of moving the class into `internal` package?
I was thinking it's to restrict the access within its module, but seems it's not?

**Comment by abhishek1508** [comment]
>qq: what's the purpose of moving the class into internal package?
I was thinking it's to restrict the access within its module, but seems it's not?

There are public classes, that have been written so that they can be used across different packages. But, these classes are not meant to be used by external developers or rather they should not depend on it. Moving it to `internal` gives them the impression that these public classes are only meant to be changed internally by the SDK and can introduce any breaking changes. They should only rely on the public API's exposed. The same approach is followed by `Square`
cc @JunDai

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java (hunk 1 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java:283]: @kmadsen Yeah this was a part of package moving change.

Comment (kmadsen) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java:283]: spoke offline, on purpose 👍

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java [hunk 2]
```diff
@@ -1,127 +0,0 @@
...
- * dynamic programming solution.
- * <p>
- * <p>
- * The running time of the Damerau-Levenshtein algorithm is O(n*m) where n is
- * the length of the source string and m is the length of the target string.
- * This implementation consumes O(n*m) space.
- *
- * @author Kevin L. Stern
- */
-class DamerauLevenshteinAlgorithm {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java:59]: Seeing this noticed that `NavigationViewRouter` was not included in https://github.com/mapbox/mapbox-navigation-android/pull/2503 "killing" this class as it's never used. Wouldn't https://github.com/mapbox/mapbox-navigation-android/issues/764 regress? 👀 https://github.com/mapbox/mapbox-navigation-android/pull/808

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java [hunk 3]
```diff
@@ -42,8 +42,9 @@
 import com.mapbox.navigation.ui.camera.DynamicCamera;
 import com.mapbox.navigation.ui.feedback.FeedbackItem;
 import com.mapbox.navigation.ui.instruction.BannerInstructionModel;
-import com.mapbox.navigation.ui.instruction.InstructionModel;
-import com.mapbox.navigation.ui.junction.RouteJunctionModel;
+import com.mapbox.navigation.ui.instruction.internal.InstructionModel;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:45]: Typo, it should be `com.mapbox.navigation.ui.internal.instruction.InstructionModel`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/BannerInstructionModel.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import com.mapbox.api.directions.v5.models.BannerText;
 import com.mapbox.navigation.base.formatter.DistanceFormatter;
 import com.mapbox.navigation.base.trip.model.RouteProgress;
+import com.mapbox.navigation.ui.instruction.internal.InstructionModel;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/BannerInstructionModel.java:10]: Same as above re: `com.mapbox.navigation.ui.internal.instruction.InstructionModel`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java [hunk 5]
```diff
@@ -55,14 +55,15 @@
 import com.mapbox.navigation.ui.feedback.FeedbackBottomSheet;
 import com.mapbox.navigation.ui.feedback.FeedbackBottomSheetListener;
 import com.mapbox.navigation.ui.feedback.FeedbackItem;
+import com.mapbox.navigation.ui.instruction.internal.InstructionModel;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java:58]: Same as above re: `com.mapbox.navigation.ui.internal.instruction.InstructionModel`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/internal/InstructionModel.java [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package com.mapbox.navigation.ui.instruction;
+package com.mapbox.navigation.ui.instruction.internal;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/internal/InstructionModel.java:1]: Same as above re: `com.mapbox.navigation.ui.internal.instruction.InstructionModel`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/TurnLaneViewHolder.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import com.mapbox.libnavigation.ui.R;
+import com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneView;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/TurnLaneViewHolder.java:8]: Typo, it should be `com.mapbox.navigation.ui.internal.instruction.turnlane.TurnLaneView`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/internal/TurnLaneDrawableMap.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import com.mapbox.libnavigation.ui.R;
+
+import java.util.HashMap;
+
+import static com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneViewData.DRAW_LANE_RIGHT;
+import static com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneViewData.DRAW_LANE_RIGHT_ONLY;
+import static com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneViewData.DRAW_LANE_SLIGHT_RIGHT;
+import static com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneViewData.DRAW_LANE_STRAIGHT;
+import static com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneViewData.DRAW_LANE_STRAIGHT_ONLY;
+import static com.mapbox.navigation.ui.instruction.turnlane.internal.TurnLaneViewData.DRAW_LANE_UTURN;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/internal/TurnLaneDrawableMap.java:12]: Same as above re: `com.mapbox.navigation.ui.internal.instruction.turnlane.`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/internal/TurnLaneView.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-package com.mapbox.navigation.ui.instruction.turnlane;
+package com.mapbox.navigation.ui.instruction.turnlane.internal;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/internal/TurnLaneView.java:1]: Same as above re: `com.mapbox.navigation.ui.internal.instruction.turnlane.`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/internal/TurnLaneViewData.java [hunk 10]
```diff
@@ -1,28 +1,28 @@
-package com.mapbox.navigation.ui.instruction.turnlane;
+package com.mapbox.navigation.ui.instruction.turnlane.internal;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/turnlane/internal/TurnLaneViewData.java:1]: Same as above re: `com.mapbox.navigation.ui.internal.instruction.turnlane.`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapFpsDelegate.java [hunk 11]
```diff
@@ -15,7 +15,7 @@
 
 class MapFpsDelegate implements OnTrackingModeChangedListener, OnTrackingModeTransitionListener {
 
-  static final int DEFAULT_MAX_FPS_THRESHOLD = 20;
+  public static final int DEFAULT_MAX_FPS_THRESHOLD = 20;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapFpsDelegate.java:18]: > Access can be package-private

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationLauncher.java [hunk 12]
```diff
@@ -1,133 +0,0 @@
...
- * <p>
- * You can launch the UI a route you have already retrieved from
- * {@link Router}.
- * </p><p>
- * For testing, you can launch with simulation, in which our
- * {@link MapboxReplayer} will begin
- * following the given {@link DirectionsRoute} once the UI is initialized.
- * </p>
- */
-public class NavigationLauncher {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationLauncher.java:26]: File was't removed 🔥 🙏

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenterTest.java [hunk 13]
```diff
@@ -1,4 +1,4 @@
-package com.mapbox.navigation.ui.summary.list;
+package com.mapbox.navigation.ui.internal.summary;
 
 import android.text.SpannableString;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenterTest.java:4]: Why most of the tests here are commented out 🤔?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapFpsDelegate.java [hunk 14]
```diff
@@ -15,7 +15,7 @@
 
 class MapFpsDelegate implements OnTrackingModeChangedListener, OnTrackingModeTransitionListener {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapFpsDelegate.java:16]: Why `MapFpsDelegateTest`s are commented out 🤔?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java [hunk 15]
```diff
@@ -20,15 +20,17 @@
 import com.mapbox.mapboxsdk.maps.Style;
 import com.mapbox.navigation.base.trip.model.RouteProgress;
 import com.mapbox.navigation.core.MapboxNavigation;
-import com.mapbox.navigation.ui.utils.CompareUtils;
+import com.mapbox.navigation.ui.internal.route.MapRouteLayerProvider;
+import com.mapbox.navigation.ui.internal.route.MapRouteSourceProvider;
+import com.mapbox.navigation.ui.internal.utils.CompareUtils;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java:26]: Why a bunch of `NavigationMapRouteTest`s are commented out 🤔?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java [hunk 16]
```diff
@@ -42,8 +42,9 @@
...
+import com.mapbox.navigation.ui.internal.ConnectivityStatusProvider;
+import com.mapbox.navigation.ui.internal.junction.RouteJunctionModel;
 import com.mapbox.navigation.ui.summary.SummaryModel;
 import com.mapbox.navigation.ui.voice.NavigationSpeechPlayer;
 import com.mapbox.navigation.ui.voice.SpeechPlayer;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:50]: Why most of `NavigationViewModelTest`s are commented out 🤔?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenterTest.java (hunk 13 above)
Comment (abhishek1508) [reply libnavigation-ui/src/test/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenterTest.java:4]: Will be taken care separately in a different ticket.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapFpsDelegate.java (hunk 14 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapFpsDelegate.java:16]: Will be taken care separately in a different ticket.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java (hunk 15 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java:26]: Will be taken care separately in a different ticket.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java (hunk 16 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:50]: Will be taken care separately in a different ticket.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java (hunk 2 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java:59]: Removing it for now as it is never used. We will do some research on if we need it.

Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java:59]: Let's cut a ticket so we don't forget 🙏

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenterTest.java (hunk 13 above)
Comment (Guardiola31337) [reply libnavigation-ui/src/test/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenterTest.java:4]: Let's cut a ticket so we don't forget 🙏

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java (hunk 2 above)
Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/DamerauLevenshteinAlgorithm.java:59]: Did we cut a ticket for this?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java (hunk 16 above)
Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java:50]: Tracking here https://github.com/mapbox/mapbox-navigation-android/pull/3104 👍

**Comment by Guardiola31337** [comment]
@abhishek1508 
> This is the only kotlin file in `Utils`. Doesn't make sense to have another `util` package outside internal just to host 1 file. Will let it be inside `internal`

We are not forced to create a package 😅 Pointed it out to be consistent across the codebase.

> - [x] Does `DefaultMapboxPuckDrawableSupplier` need to be public?
> - [x] Does `NavigationPuckPresenter` need to be public?

Maybe now that both `DefaultMapboxPuckDrawableSupplier` and `NavigationPuckPresenter` are package-private we can extract `puck` package out of the `internal`

> - [ ] Does `MobileNetworkChecker` need to be under `internal`? It is `internal` already
> Yes. It is package private. But `ConnectivityStatusProvider` is `public` and lives inside `internal` package. `ConnectivityStatusProvider` uses `MobileNetworkChecker` and hence it needs to be live in the same package as `ConnectivityStatusProvider` or be made public.
> - [ ] Does `WifiNetworkChecker` need to be under `internal`? It is `internal` already
> Yes. It is package private. But `ConnectivityStatusProvider` is `public` and lives inside `internal` package. `ConnectivityStatusProvider` uses `WifiNetworkChecker` and hence it needs to be live in the same package as `ConnectivityStatusProvider` or be made public.

👍 

> - [ ] Does the `NavigationViewModel` need to be public?
> Moving this to internal will result in a whole lot files being changed. It is not worth going through that amount of changes. Will leave it as is.

👍 

>  - [ ] Why all the listeners under internal.listeners are internal?

These seem that need to be exposed publicly. Am I missing something?
